### PR TITLE
BUG-97056 Introduce VMTypes test cases and support classes

### PR DESCRIPTION
**Pull request by aszegedi**
@afarsang Please review new API tests for VM Types Service.

### CODE REVIEW

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/RegionTests.java [hunk 1]
```diff
@@ -160,7 +161,7 @@ public void testListAvailabilityZonesForDefaultRegion() throws Exception {
                             .get(cloudProvider.region());
 
                     if (cloudProvider.getPlatform().equalsIgnoreCase("AZURE")) {
```
Comment (afarsang) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/RegionTests.java]: The conditional part could be another test, probably platform dependent. In general a test should be as straightforward as possible, any condition in a test case means another test case; common parts could be in BeforeTest or other method.

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/VmTypeTests.java [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+                .withAvailibilityZone(cloudProvider.availibilityZone()), cloudProvider.getPlatform()
+                .concat(" vm type request"));
+        when(VmType.getPlatformVmTypes(), " vm types are requested to "
+                .concat(cloudProvider.getPlatform())
+                .concat(" credential and ")
+                .concat(cloudProvider.region())
+                .concat(" region."));
+        then(VmType.assertThis(
+                (vmtype, t) -> {
+                    if (cloudProvider.getPlatform().matches("AZURE")) {
```
Comment (afarsang) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/VmTypeTests.java]: ... in two tests, please

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/VmTypeTests.java [hunk 3]
```diff
@@ -0,0 +1,217 @@
+package com.sequenceiq.it.cloudbreak;
```
Comment (mhmxs) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/VmTypeTests.java]: Could you please turn on Idea inspections and fix all possible code smells?

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/VmTypeAction.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import java.io.IOException;
+
+public class VmTypeAction {
+
+    private VmTypeAction() {
+    }
+
+    public static void getVmTypesByCredentialId(IntegrationTestContext integrationTestContext, Entity entity) throws IOException {
+        VmTypeEntity vmtypeEntity = (VmTypeEntity) entity;
+        CloudbreakClient client;
```
Comment (mhmxs) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/VmTypeAction.java]: Should be one line the declaration and the initialization

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/cloud/AwsCloudProvider.java [hunk 5]
```diff
@@ -70,6 +70,14 @@ public String region() {
         return regionParam == null ? region : regionParam;
     }
 
+    @Override
+    public String availibilityZone() {
```
Comment (mhmxs) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/cloud/AwsCloudProvider.java]: Code duplication expect input parameters.

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/VmTypeTests.java [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.ForbiddenException;
+import java.util.HashSet;
+import java.util.Set;
+
+public class VmTypeTests extends CloudbreakTest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(VmTypeTests.class);
+
+    private String errorMessage = "";
```
Comment (mhmxs) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/VmTypeTests.java]: Error message should be a local variable

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/VmTypeTests.java [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+
+    private CloudProvider cloudProvider;
+
+    private String regionDefVirtualMachine = "";
```
Comment (mhmxs) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/VmTypeTests.java]: Should be a local variable

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/VmTypeTests.java [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+
+    private String azDefVirtualMachine = "";
```
Comment (mhmxs) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/VmTypeTests.java]: Should be a local variable

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/VmTypeTests.java [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+
+    private int regionVirtualMachinesNumber;
```
Comment (mhmxs) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/VmTypeTests.java]: Should be a local variable

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/VmTypeTests.java [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+
+    private int azVirtualMachinesNumber;
```
Comment (mhmxs) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/VmTypeTests.java]: Should be a local variable

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/VmTypeTests.java [hunk 11]
```diff
@@ -0,0 +1,217 @@
...
+
+    private Set<VmTypeJson> regionVirtualMachines = new HashSet<VmTypeJson>();
+
+    private Set<VmTypeJson> azVirtualMachines = new HashSet<VmTypeJson>();
+
+    public VmTypeTests() {
```
Comment (mhmxs) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/VmTypeTests.java]: Default empty constructor is not used

**Comment by aszegedi** [comment]
@mhmxs and @afarsang Please wait with the review and merge till notification, code fixes are ongoing.

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/VmTypeTests.java (hunk 3 above)
Comment (aszegedi) [reply integration-test/src/main/java/com/sequenceiq/it/cloudbreak/VmTypeTests.java]: Thank you for your hint on this. This has been turned on and regularly used.

Comment (mhmxs) [reply integration-test/src/main/java/com/sequenceiq/it/cloudbreak/VmTypeTests.java]: (y)

**Comment by aszegedi** [comment]
@afarsang and @mhmxs I've done with the needed fixes in classes. Please review again in your first convenience.

**Pull request merged**